### PR TITLE
8342644: Optimize InvokerBytecodeGenerator for using ClassFile API

### PR DESCRIPTION
**Pull request by wenshao**
Cache commonly used classDesc to avoid creating ReferenceClassDescImpl every time

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8342644](https://bugs.openjdk.org/browse/JDK-8342644): Optimize InvokerBytecodeGenerator for using ClassFile API (**Enhancement** - P4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21580/head:pull/21580` \
`$ git checkout pull/21580`

Update a local copy of the PR: \
`$ git checkout pull/21580` \
`$ git pull https://git.openjdk.org/jdk.git pull/21580/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21580`

View PR using the GUI difftool: \
`$ git pr show -t 21580`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21580.diff">https://git.openjdk.org/jdk/pull/21580.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21580#issuecomment-2423482422)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java [hunk 1]
```diff
@@ -131,9 +135,17 @@ private InvokerBytecodeGenerator(LambdaForm lambdaForm, int localsMapSize,
             name = makeDumpableClassName(name);
         }
         this.name = name;
-        this.className = CLASS_PREFIX.concat(name);
-        validateInternalClassName(name);
-        this.classEntry = pool.classEntry(ReferenceClassDescImpl.ofValidated(concat("L", className, ";")));
+        var classDesc = switch (name) {
+            case "MH"  -> CD_MH;
+            case "DMH" -> CD_DMH;
+            case "BMH" -> CD_BMH;
```
Comment (liach) [inline src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java:141]: I don't see where DMH and BMH are passed; only see NFI and LFI.

Comment (wenshao) [reply src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java:141]: This DMH is derived from invokerName, which comes from java.lang.invoke.LambdaForm.Kind#defaultLambdaName

Comment (ExE-Boss) [reply src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java:141]: `"NFI"` and `"LFI"` should probably also be added to this `switch`.

Comment (cl4es) [reply src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java:143]: No: I believe these names are used for invokers spun exclusively for the LF interpreter, which is only used for debugging (and as a development aide) these days. Micro-optimizing those code paths only add complexity for no gain in practice.

**Comment by liach** [comment]
Would it make sense for us to pass the ClassDesc as a constructor argument? We can drop the SourceFile attribute as I think we included it by convenience back when we were using ASM.

**Pull request closed without merging**